### ISSUE TITLE
Error loading in ROS Melodic

### ISSUE THREAD

**Issue by UPedrou** [aws-robotics/aws-robomaker-hospital-world#10]
Hi,
I am trying to load the hospital world in gazebo, I can compile everything correctly.
But when I execute roslaunch view_hospital.launch, I have the following errors in the terminal and the world launches correctly but some models are missing.
```
Error Code 11 Msg: Unable to find uri[model://IVStand]
Error Code 11 Msg: Unable to find uri[model://BMWCart]
Error Code 11 Msg: Unable to find uri[model://Scrubs]
Error Code 11 Msg: Unable to find uri[model://BPCart]
Error Code 11 Msg: Unable to find uri[model://BPCart]
Error Code 11 Msg: Unable to find uri[model://BPCart]
Error Code 11 Msg: Unable to find uri[model://BPCart]
Error Code 11 Msg: Unable to find uri[model://VendingMachine]
Error Code 11 Msg: Unable to find uri[model://VendingMachine]
Error Code 11 Msg: Unable to find uri[model://Chair]
[Msg] Loading world file [/root/COBOTTA/catkin_ws/src/aws-robomaker-hospital-world/worlds/hospital.world]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[ INFO] [1622092038.499380288]: waitForService: Service [/gazebo/set_physics_properties] is now available.
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[ INFO] [1622092038.508959044]: Physics dynamic reconfigure ready.
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [ColladaLoader.cc:1537] Unable to add material[]
[Wrn] [Event.cc:61] Warning: Deleting a connection right after creation. Make sure to save the ConnectionPtr from a Connect call
```

I am using ROS Melodic and Gazebo 9.18, this is a image of the environment loaded [https://imgur.com/a/shytdIr](https://imgur.com/a/shytdIr)

Does anyone has any recommendation in how to solve this issue? 
Thank you

**Comment by ojasjoshi** [comment]
Hi, can you please share what exact steps you ran to compile and build the world package?

The world definition depends on a few models from ignition and these models are downloaded [here](https://github.com/aws-robotics/aws-robomaker-hospital-world/blob/master/CMakeLists.txt#L9). My hunch is that some of the ignition models failed to download during `colcon build` and hence are not available to gazebo to load.

I would suggest that you try loading the world directly through Gazebo. You can follow [these README instructions](https://github.com/aws-robotics/aws-robomaker-hospital-world#load-directly-into-gazebo-without-ros). If the issue is indeed with fetching/downloading the ignition models, this should also fail. Please let us know what error you see, if any.

**Comment by UPedrou** [comment]
Hi, thank you very much for your answer.

The steps that I followed were `colcon build` then `catkin build` for my other packages. After that I was launching the roslaunch `view_hospital.launch`.

I tried repeating the steps again, but now in `colcon build` I have this error message
```
Starting >>> aws_robomaker_hospital_world
--- stderr: aws_robomaker_hospital_world
CMake Error at cmake_install.cmake:56 (file):
  file INSTALL cannot find
  "/root/COBOTTA/catkin_ws/src/aws-robomaker-hospital-world/fuel_models".

make: *** [install] Error 1
---
Failed   <<< aws_robomaker_hospital_world [0.25s, exited with code 2]

Summary: 0 packages finished [0.34s]
  1 package failed: aws_robomaker_hospital_world
  1 package had stderr output: aws_robomaker_hospital_world
```
And following the steps for loading directly the world through Gazebo. I have the following error when I run the instruction `./setup.sh`
```
Traceback (most recent call last):
  File "fuel_utility.py", line 15, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'
```
I think I am missing some of the libraries required to run `fuel_utility.py`, I will install them manually and try again.

Thank you for your help.

**Comment by UPedrou** [comment]
Thank you for your answer.
The steps that I followed to compile and build the world were `colcon build` and then sourcing.

I followed the instructions to load the world directly through Gazebo. Initially I had errors when running `setup.sh`. The errors were of missing libraries such as `requests`,`docopt` and `lxml`. I installed manually those libraries and ran again `setup.sh` and everything worked and I could launch the gazebo with the world.

I think that the error was related to my installation of Python3 and Colcon.
Thank you for your help.
